functests/2_performance_update/updating_profile.go

```diff
@@ -64,30 +66,56 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			isolatedCPUSet, err := cpuset.Parse(string(*profile.Spec.CPU.Isolated))
 			Expect(err).ToNot(HaveOccurred())
 
-			var expectedBannedCPUs cpuset.CPUSet
-			if irqLoadBalancingDisabled {
-				expectedBannedCPUs = isolatedCPUSet
-			} else {
-				expectedBannedCPUs = cpuset.NewCPUSet()
-			}
+			verifyNodes := func() error {
+				var expectedBannedCPUs cpuset.CPUSet
+				if irqLoadBalancingDisabled {
+					expectedBannedCPUs = isolatedCPUSet
+				} else {
+					expectedBannedCPUs = cpuset.NewCPUSet()
+				}
 
-			for _, node := range workerRTNodes {
-				bannedCPUs, err := nodes.BannedCPUs(node)
-				Expect(err).ToNot(HaveOccurred(), fmt.Errorf("failed to extract the banned CPUs from node %s: %v", node.Name, err))
-				Expect(bannedCPUs.Equals(expectedBannedCPUs)).To(BeTrue(),
-					fmt.Errorf("banned CPUs %v do not match the expected mask %v on node %s",
-						bannedCPUs, expectedBannedCPUs, node.Name))
+				for _, node := range workerRTNodes {
+					bannedCPUs, err := nodes.BannedCPUs(node)
+					Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to extract the banned CPUs from node %s", node.Name))
+
+					if !bannedCPUs.Equals(expectedBannedCPUs) {
+						return fmt.Errorf("banned CPUs %v do not match the expected mask %v on node %s",
+							bannedCPUs, expectedBannedCPUs, node.Name)
+					}
+					smpAffinity, err := nodes.ExecCommandOnNode(chkDefaultSMPAffinity, &node)
+					Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to execute %s", chkDefaultSMPAffinity))
+
+					smpAffinitySet, err := components.CPUMaskToCPUSet(smpAffinity)
+					Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to parse %s", smpAffinity))
+
+					onlineCPUs, err := nodes.ExecCommandOnNode(chkOnlineCPUs, &node)
+					Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to execute %s", chkOnlineCPUs))
+
+					onlineCPUsSet, err := cpuset.Parse(onlineCPUs)
+					Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to parse %s", onlineCPUs))
+
+					if irqLoadBalancingDisabled {
+						if !smpAffinitySet.Equals(onlineCPUsSet.Difference(isolatedCPUSet)) {
+							return fmt.Errorf("found Default SMP Affinity %v, expected %v",
+								onlineCPUs, onlineCPUsSet.Difference(isolatedCPUSet))
+						}
+					} else {
+						if !smpAffinitySet.Equals(onlineCPUsSet) {
+							return fmt.Errorf("found Default SMP Affinity %v, expected %v",
+								onlineCPUs, onlineCPUsSet.Difference(isolatedCPUSet))
```

Comment:
hm, is it typo and these two should be switched?

Reply:
Is not a type, this is what I test:
1. irqLoadBalancingDisabled -> default_smp_affinity should be only  "non-isolated" CPUs
2. irqLoadBalancingEnabled -> default_smp_affinity should be all online cpus

Reply:
brr, i'm slightly confused.. 
I'm talking about error message, what I see:

if smpAffinitySet != $onlineCPUS --- show message: "found Default SMP Affinity $onlineCPUS, expected $difference

but I guess it should be: found $difference, expected $onlineCPUS

Reply:
the logic itself right, i'm talking about error message only

Reply:
Aha, you are right! Nice catch.

Reply:
no any complains other than that, so let's unhold PR when you fix this issue?

Reply:
done, since the code is re-pushed, can you please lgtm again?

Reply:
/lgtm

Reply:
thanks!!